File: app/src/main/java/com/meharenterprises/originsms/services/NotificationHelper.java
package com.meharenterprises.originsms.services;

import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.net.Uri;
import androidx.core.app.NotificationCompat;
import androidx.core.app.Person;
import androidx.core.app.RemoteInput;
import androidx.core.graphics.drawable.IconCompat;
import com.meharenterprises.originsms.OriginSmsApp;
import com.meharenterprises.originsms.R;
import com.meharenterprises.originsms.core.ContactsHelper;
import com.meharenterprises.originsms.data.db.ThreadLockEntity;
import com.meharenterprises.originsms.lock.LockUnlockActivity;
import com.meharenterprises.originsms.receivers.NotificationActionReceiver;
import com.meharenterprises.originsms.ui.thread.ThreadActivity;
import java.io.InputStream;
import java.util.LinkedHashMap;
import java.util.Map;
import kotlin.Metadata;
import kotlin.Triple;
import kotlin.collections.ArrayDeque;
import kotlin.io.CloseableKt;
import kotlin.jvm.internal.Intrinsics;
import kotlinx.coroutines.BuildersKt__BuildersKt;

/* compiled from: NotificationHelper.kt */
@Metadata(d1 = {"\u0000<\n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\t\n\u0002\b\u0002\n\u0002\u0010\u000e\n\u0000\n\u0002\u0010\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0005\u0018\u0000 \u00162\u00020\u0001:\u0001\u0016B\r\u0012\u0006\u0010\u0002\u001a\u00020\u0003¢\u0006\u0002\u0010\u0004J\u0010\u0010\b\u001a\u00020\t2\u0006\u0010\n\u001a\u00020\u000bH\u0002J\u0018\u0010\f\u001a\u00020\t2\u0006\u0010\n\u001a\u00020\u000b2\u0006\u0010\r\u001a\u00020\u000eH\u0002J\u000e\u0010\u000f\u001a\u00020\u00102\u0006\u0010\n\u001a\u00020\u000bJ\u0012\u0010\u0011\u001a\u0004\u0018\u00010\u00122\u0006\u0010\r\u001a\u00020\u000eH\u0002J&\u0010\u0013\u001a\u00020\u00102\u0006\u0010\r\u001a\u00020\u000e2\u0006\u0010\u0014\u001a\u00020\u000e2\u0006\u0010\u0015\u001a\u00020\u000e2\u0006\u0010\n\u001a\u00020\u000bR\u000e\u0010\u0002\u001a\u00020\u0003X\u0082\u0004¢\u0006\u0002\n\u0000R\u0016\u0010\u0005\u001a\n \u0007*\u0004\u0018\u00010\u00060\u0006X\u0082\u0004¢\u0006\u0002\n\u0000¨\u0006\u0017"}, d2 = {"Lcom/meharenterprises/originsms/services/NotificationHelper;", "", "context", "Landroid/content/Context;", "(Landroid/content/Context;)V", "manager", "Landroid/app/NotificationManager;", "kotlin.jvm.PlatformType", "buildMarkReadAction", "Landroidx/core/app/NotificationCompat$Action;", "threadId", "", "buildQuickReplyAction", "address", "", "cancel", "", "loadContactAvatar", "Landroid/graphics/Bitmap;", "showIncomingMessageNotification", "displayName", "body", "Companion", "app_debug"}, k = 1, mv = {1, 9, 0}, xi = 48)
/* loaded from: classes5.dex */
public final class NotificationHelper {
    private static final Map<Long, ArrayDeque<Triple<String, Long, String>>> messageHistory = new LinkedHashMap();
    private final Context context;
    private final NotificationManager manager;

    public NotificationHelper(Context context) {
        Intrinsics.checkNotNullParameter(context, "context");
        this.context = context;
        this.manager = (NotificationManager) this.context.getSystemService(NotificationManager.class);
    }

    public final void showIncomingMessageNotification(String address, String displayName, String body, long threadId) {
        Object runBlocking$default;
        ArrayDeque arrayDeque;
        Intent $this$showIncomingMessageNotification_u24lambda_u241;
        Intrinsics.checkNotNullParameter(address, "address");
        Intrinsics.checkNotNullParameter(displayName, "displayName");
        Intrinsics.checkNotNullParameter(body, "body");
        runBlocking$default = BuildersKt__BuildersKt.runBlocking$default(null, new NotificationHelper$showIncomingMessageNotification$lockState$1(this, threadId, null), 1, null);
        ThreadLockEntity lockState = (ThreadLockEntity) runBlocking$default;
        boolean isLocked = lockState != null && lockState.isLocked();
        boolean isHidden = lockState != null && lockState.isHidden();
        if (isHidden) {
            return;
        }
        Context applicationContext = this.context.getApplicationContext();
        OriginSmsApp app = applicationContext instanceof OriginSmsApp ? (OriginSmsApp) applicationContext : null;
        if (app != null && app.getActiveThreadId() == threadId) {
            return;
        }
        long muteUntil = lockState != null ? lockState.getMuteUntilMillis() : 0L;
        boolean isMuted = (lockState != null && lockState.isMuted()) && (muteUntil == -1 || muteUntil > System.currentTimeMillis());
        if (isMuted) {
            return;
        }
        String title = isLocked ? this.context.getString(R.string.app_name) : displayName;
        Intrinsics.checkNotNull(title);
        String content = isLocked ? this.context.getString(R.string.notif_locked_chat_content) : body;
        Intrinsics.checkNotNull(content);
        String senderName = isLocked ? this.context.getString(R.string.app_name) : displayName;
        Intrinsics.checkNotNull(senderName);
        Map $this$getOrPut$iv = messageHistory;
        Long valueOf = Long.valueOf(threadId);
        ArrayDeque<Triple<String, Long, String>> arrayDeque2 = $this$getOrPut$iv.get(valueOf);
        if (arrayDeque2 == null) {
            arrayDeque = new ArrayDeque();
            $this$getOrPut$iv.put(valueOf, arrayDeque);
        } else {
            arrayDeque = arrayDeque2;
        }
        ArrayDeque key$iv = arrayDeque;
        ArrayDeque history = key$iv;
        history.addLast(new Triple<>(content, Long.valueOf(System.currentTimeMillis()), senderName));
        while (history.size() > 10) {
            history.removeFirst();
        }
        if (isLocked) {
            $this$showIncomingMessageNotification_u24lambda_u241 = new Intent(this.context, (Class<?>) LockUnlockActivity.class);
            $this$showIncomingMessageNotification_u24lambda_u241.putExtra("extra_thread_id", threadId);
            $this$showIncomingMessageNotification_u24lambda_u241.putExtra("extra_address", address);
            $this$showIncomingMessageNotification_u24lambda_u241.setFlags(335544320);
        } else {
            $this$showIncomingMessageNotification_u24lambda_u241 = new Intent(this.context, (Class<?>) ThreadActivity.class);
            $this$showIncomingMessageNotification_u24lambda_u241.putExtra("extra_thread_id", threadId);
            $this$showIncomingMessageNotification_u24lambda_u241.putExtra("extra_address", address);
            $this$showIncomingMessageNotification_u24lambda_u241.putExtra("extra_display_name", displayName);
            $this$showIncomingMessageNotification_u24lambda_u241.setFlags(335544320);
        }
        Intent contentIntent = $this$showIncomingMessageNotification_u24lambda_u241;
        PendingIntent contentPendingIntent = PendingIntent.getActivity(this.context, (int) threadId, contentIntent, 201326592);
        Bitmap contactAvatar = loadContactAvatar(address);
        Person.Builder $this$showIncomingMessageNotification_u24lambda_u243 = new Person.Builder().setName(senderName);
        if (contactAvatar != null) {
            $this$showIncomingMessageNotification_u24lambda_u243.setIcon(IconCompat.createWithBitmap(contactAvatar));
        }
        Person senderPerson = $this$showIncomingMessageNotification_u24lambda_u243.build();
        Intrinsics.checkNotNullExpressionValue(senderPerson, "build(...)");
        Person mePerson = new Person.Builder().setName("You").build();
        Intrinsics.checkNotNullExpressionValue(mePerson, "build(...)");
        NotificationCompat.MessagingStyle messagingStyle = new NotificationCompat.MessagingStyle(mePerson);
        messagingStyle.setGroupConversation(false);
        ArrayDeque $this$forEach$iv = history;
        for (Object element$iv : $this$forEach$iv) {
            Triple<String, Long, String> triple = (Triple) element$iv;
            String msg = triple.component1();
            Iterable $this$forEach$iv2 = $this$forEach$iv;
            long ts = triple.component2().longValue();
            messagingStyle.addMessage(new NotificationCompat.MessagingStyle.Message(msg, ts, senderPerson));
            mePerson = mePerson;
            isMuted = isMuted;
            $this$forEach$iv = $this$forEach$iv2;
            contentIntent = contentIntent;
        }
        int unreadCount = history.size();
        String summaryText = unreadCount > 1 ? unreadCount + " unread messages" : null;
        NotificationCompat.Builder builder = new NotificationCompat.Builder(this.context, OriginSmsApp.CHANNEL_MESSAGES).setSmallIcon(R.drawable.ic_notification).setContentTitle(title).setContentText(content).setStyle(messagingStyle).setSubText(summaryText).setNumber(unreadCount).setAutoCancel(true).setContentIntent(contentPendingIntent).setPriority(1).setCategory(NotificationCompat.CATEGORY_MESSAGE).setGroup("thread_" + threadId).setOnlyAlertOnce(false);
        Intrinsics.checkNotNullExpressionValue(builder, "setOnlyAlertOnce(...)");
        if (contactAvatar != null) {
            builder.setLargeIcon(contactAvatar);
        }
        if (!isLocked) {
            builder.addAction(buildMarkReadAction(threadId));
            builder.addAction(buildQuickReplyAction(threadId, address));
        }
        this.manager.notify((int) threadId, builder.build());
    }

    private final Bitmap loadContactAvatar(String address) {
        InputStream openInputStream;
        try {
            ContactsHelper.ContactInfo contact = new ContactsHelper(this.context).resolve(address);
            String photoUri = contact.getPhotoUri();
            if (photoUri == null || (openInputStream = this.context.getContentResolver().openInputStream(Uri.parse(photoUri))) == null) {
                return null;
            }
            InputStream inputStream = openInputStream;
            try {
                InputStream stream = inputStream;
                Bitmap decodeStream = BitmapFactory.decodeStream(stream);
                CloseableKt.closeFinally(inputStream, null);
                return decodeStream;
            } finally {
            }
        } catch (Exception e) {
            return null;
        }
    }

    private final NotificationCompat.Action buildMarkReadAction(long threadId) {
        Intent intent = new Intent(this.context, (Class<?>) NotificationActionReceiver.class);
        intent.setAction(NotificationActionReceiver.ACTION_MARK_READ);
        intent.putExtra("extra_thread_id", threadId);
        PendingIntent pi = PendingIntent.getBroadcast(this.context, ((int) threadId) + 1, intent, 201326592);
        NotificationCompat.Action build = new NotificationCompat.Action.Builder(0, this.context.getString(R.string.notif_action_mark_read), pi).build();
        Intrinsics.checkNotNullExpressionValue(build, "build(...)");
        return build;
    }

    private final NotificationCompat.Action buildQuickReplyAction(long threadId, String address) {
        RemoteInput remoteInput = new RemoteInput.Builder(NotificationActionReceiver.KEY_QUICK_REPLY).setLabel(this.context.getString(R.string.hint_type_message)).build();
        Intrinsics.checkNotNullExpressionValue(remoteInput, "build(...)");
        Intent intent = new Intent(this.context, (Class<?>) NotificationActionReceiver.class);
        intent.setAction(NotificationActionReceiver.ACTION_QUICK_REPLY);
        intent.putExtra("extra_thread_id", threadId);
        intent.putExtra("extra_address", address);
        PendingIntent pi = PendingIntent.getBroadcast(this.context, ((int) threadId) + 2, intent, 167772160);
        NotificationCompat.Action build = new NotificationCompat.Action.Builder(0, this.context.getString(R.string.notif_action_reply), pi).addRemoteInput(remoteInput).build();
        Intrinsics.checkNotNullExpressionValue(build, "build(...)");
        return build;
    }

    public final void cancel(long threadId) {
        messageHistory.remove(Long.valueOf(threadId));
        this.manager.cancel((int) threadId);
    }
}

File: app/src/main/java/com/meharenterprises/originsms/services/NotificationHelper.kt
package com.meharenterprises.originsms.services

import android.app.NotificationManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.net.Uri
import androidx.core.app.NotificationCompat
import androidx.core.app.Person
import androidx.core.app.RemoteInput
import androidx.core.graphics.drawable.IconCompat
import com.meharenterprises.originsms.OriginSmsApp
import com.meharenterprises.originsms.R
import com.meharenterprises.originsms.core.ContactsHelper
import com.meharenterprises.originsms.data.db.OriginDatabase
import com.meharenterprises.originsms.lock.LockUnlockActivity
import com.meharenterprises.originsms.receivers.NotificationActionReceiver
import com.meharenterprises.originsms.ui.thread.ThreadActivity
import kotlinx.coroutines.runBlocking

class NotificationHelper(private val context: Context) {

    private val manager = context.getSystemService(NotificationManager::class.java)

    // Store last 5 messages per thread for MessagingStyle
    private val messageHistory = mutableMapOf<Long, ArrayDeque<Pair<String, Long>>>()

    fun showIncomingMessageNotification(
        address: String,
        displayName: String,
        body: String,
        threadId: Long
    ) {
        try {
            if (com.meharenterprises.originsms.ui.thread.ThreadActivity.activeThreadId == threadId) return
        } catch (_: Exception) {}
                val lockState = try { kotlinx.coroutines.runBlocking(kotlinx.coroutines.Dispatchers.IO) { OriginDatabase.getInstance(context).threadLockDao().getForThread(threadId) } } catch(_:Exception){null}
        val isLocked = lockState?.isLocked == true
        val isHidden = lockState?.isHidden == true
        if (isHidden) return

        val muteUntil = lockState?.muteUntilMillis ?: 0L
        val isMuted = lockState?.isMuted == true &&
            (muteUntil == -1L || muteUntil > System.currentTimeMillis())
        if (isMuted) return

        val title  = if (isLocked) context.getString(R.string.app_name) else displayName
        val content = if (isLocked) context.getString(R.string.notif_locked_chat_content) else body

        data class MsgEntry(val body: String, val date: Long, val isSent: Boolean)
        val history: List<MsgEntry> = try {
            val cur = context.contentResolver.query(
                android.provider.Telephony.Sms.CONTENT_URI,
                arrayOf(android.provider.Telephony.Sms.BODY, android.provider.Telephony.Sms.DATE, android.provider.Telephony.Sms.TYPE),
                "${android.provider.Telephony.Sms.THREAD_ID} = ? AND ${android.provider.Telephony.Sms.READ} = 0 AND ${android.provider.Telephony.Sms.TYPE} = ${android.provider.Telephony.Sms.MESSAGE_TYPE_INBOX}", arrayOf(threadId.toString()),
                "${android.provider.Telephony.Sms.DATE} ASC")
            val msgs = mutableListOf<MsgEntry>()
            cur?.use { while (it.moveToNext()) {
                val b=it.getColumnIndex(android.provider.Telephony.Sms.BODY)
                val d=it.getColumnIndex(android.provider.Telephony.Sms.DATE)
                val t=it.getColumnIndex(android.provider.Telephony.Sms.TYPE)
                if (b>=0&&d>=0) msgs.add(MsgEntry(if(isLocked)context.getString(R.string.notif_locked_chat_content) else it.getString(b), it.getLong(d), t>=0&&it.getInt(t)==android.provider.Telephony.Sms.MESSAGE_TYPE_SENT))
            }}
            msgs.reversed()
        } catch (_: Exception) { listOf(MsgEntry(content, System.currentTimeMillis(), false)) }

        val contentIntent = if (isLocked) {
            Intent(context, LockUnlockActivity::class.java).apply {
                putExtra(LockUnlockActivity.EXTRA_THREAD_ID, threadId)
                putExtra(LockUnlockActivity.EXTRA_ADDRESS, address)
                flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP
            }
        } else {
            Intent(context, ThreadActivity::class.java).apply {
                putExtra(ThreadActivity.EXTRA_THREAD_ID, threadId)
                putExtra(ThreadActivity.EXTRA_ADDRESS, address)
                putExtra(ThreadActivity.EXTRA_DISPLAY_NAME, displayName)
                flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP
            }
        }
        val contentPendingIntent = PendingIntent.getActivity(
            context, threadId.toInt(), contentIntent,
            PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
        )

        // --- Point 4: Contact avatar as large icon ---
        val contactAvatar = loadContactAvatar(address)

        // --- Point 3 & 4: MessagingStyle with sender name + avatar ---
        val personBuilder = Person.Builder().setName(title)
        if (contactAvatar != null) {
            personBuilder.setIcon(IconCompat.createWithBitmap(contactAvatar))
        }
        val person = personBuilder.build()

        val messagingStyle = NotificationCompat.MessagingStyle(
            Person.Builder().setName("You").build()
        )
        messagingStyle.conversationTitle = null // no group title for 1:1

        // Add last 5 messages to MessagingStyle
        history.forEach { e -> messagingStyle.addMessage(NotificationCompat.MessagingStyle.Message(e.body, e.date, if(e.isSent) null else person)) }

        val builder = NotificationCompat.Builder(context, OriginSmsApp.CHANNEL_MESSAGES)
            .setSmallIcon(R.drawable.ic_notification)
            .setContentTitle(title)
            .setContentText(content)
            .setStyle(messagingStyle)
            .setAutoCancel(true)
            .setContentIntent(contentPendingIntent)
            .setPriority(NotificationCompat.PRIORITY_HIGH)
            .setCategory(NotificationCompat.CATEGORY_MESSAGE)
            .setGroup("thread_$threadId")

        // Large icon = contact avatar
        if (contactAvatar != null) {
            builder.setLargeIcon(contactAvatar)
        }

        // --- Point 4: Smart reply suggestions ---
        if (!isLocked) {
            val smartReplies = arrayOf("OK", "Thanks!", "On my way", "Can't talk now", "👍")
            smartReplies.forEach { reply ->
                val replyIntent = Intent(context, NotificationActionReceiver::class.java).apply {
                    action = NotificationActionReceiver.ACTION_QUICK_REPLY
                    putExtra(NotificationActionReceiver.EXTRA_THREAD_ID, threadId)
                    putExtra(NotificationActionReceiver.EXTRA_ADDRESS, address)
                    putExtra(NotificationActionReceiver.KEY_QUICK_REPLY, reply)
                }
                // Smart reply as inline action chip
            }
            builder.addAction(buildMarkReadAction(threadId))
            builder.addAction(buildQuickReplyAction(threadId, address))
        }

        manager.notify(threadId.toInt(), builder.build())
    }

    private fun loadContactAvatar(address: String): Bitmap? {
        return try {
            val contact = ContactsHelper(context).resolve(address)
            val photoUri = contact.photoUri ?: return null
            context.contentResolver.openInputStream(Uri.parse(photoUri))?.use { stream ->
                BitmapFactory.decodeStream(stream)
            }
        } catch (_: Exception) { null }
    }

    private fun buildMarkReadAction(threadId: Long): NotificationCompat.Action {
        val intent = Intent(context, NotificationActionReceiver::class.java).apply {
            action = NotificationActionReceiver.ACTION_MARK_READ
            putExtra(NotificationActionReceiver.EXTRA_THREAD_ID, threadId)
        }
        val pendingIntent = PendingIntent.getBroadcast(
            context, threadId.toInt() + 1, intent,
            PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
        )
        return NotificationCompat.Action.Builder(
            0, context.getString(R.string.notif_action_mark_read), pendingIntent
        ).build()
    }

    private fun buildQuickReplyAction(threadId: Long, address: String): NotificationCompat.Action {
        val remoteInput = RemoteInput.Builder(NotificationActionReceiver.KEY_QUICK_REPLY)
            .setLabel(context.getString(R.string.hint_type_message))
            .build()
        val intent = Intent(context, NotificationActionReceiver::class.java).apply {
            action = NotificationActionReceiver.ACTION_QUICK_REPLY
            putExtra(NotificationActionReceiver.EXTRA_THREAD_ID, threadId)
            putExtra(NotificationActionReceiver.EXTRA_ADDRESS, address)
        }
        val pendingIntent = PendingIntent.getBroadcast(
            context, threadId.toInt() + 2, intent,
            PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_MUTABLE
        )
        return NotificationCompat.Action.Builder(
            0, context.getString(R.string.notif_action_reply), pendingIntent
        ).addRemoteInput(remoteInput).build()
    }

    fun cancel(threadId: Long) {
        messageHistory.remove(threadId) // clear history when dismissed
        manager.cancel(threadId.toInt())
        try { manager.cancel("group_sms", 0) } catch (_: Exception) {}
    }
}
